packages/vscode-extension/src/webview/components/DeviceSettingsDropdown.tsx

```diff
@@ -97,12 +100,49 @@ function DeviceSettingsDropdown({ children, disabled }: DeviceSettingsDropdownPr
 
   const deviceSettings = use$(store$.workspaceConfiguration.deviceSettings);
 
+  const licenseStatus = use$(store$.license.status);
+  const hasProAccess = hasAccessToProFeatures(licenseStatus);
+
   const { project } = useProject();
 
   const { openModal } = useModal();
 
+  const { openPaywall } = usePaywall();
+
   const resetOptions = platform === "iOS" ? resetOptionsIOS : resetOptionsAndroid;
 
+  const handleRotateDevice = async (direction: DeviceRotationDirection) => {
+    if (!hasProAccess) {
+      openPaywall();
+      return;
+    }
+
+    try {
+      await project.rotateDevices(direction);
+    } catch (error) {
+      if (error instanceof RestrictedFunctionalityError) {
+        openPaywall();
+      }
+    }
```

Comment:
I wonder -- could we put this inside the `project` proxy returned from `useProject`?
Basically, wrap each method on the `project` object in this try/catch, so that when you _call_ the functions, you cannot forget to handle the paywall?